feat: `atlas pull` for XBlock translations | FC-0012 (#33698)
openedx/edx-platform: docs/decisions/0019-oep-58-atlas-translations-design.rst

                └── django.po

Using XBlock python module names instead of repository names
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

There's more than one identifier for XBlocks and Plugins:

#. **The XBlock/plugin tag:** Python plugins has an entry point name in which
   is referred to as ``tag`` in Open edX. For example, the

>    the tag the `Drag and Drop XBlock setup.py file`_ is ``drag-and-drop-v2``::

Reviewer: ```suggestion
#. **The XBlock/plugin tag:** Python plugins have an entry point name which
   is referred to as ``tag`` in Open edX. For example, the
   tag in the `Drag and Drop XBlock setup.py file`_ is ``drag-and-drop-v2``::
```

     # xblock-drag-and-drop-v2/setup.py
     entry_points={
         'xblock.v1': 'drag-and-drop-v2 = drag_and_drop_v2:DragAndDropBlock',
     }

#. **The git repository name:** Each XBlock has a unique git repository name.
   For example, the Drag and Drop XBlock has the ``xblock-drag-and-drop-v2``
   repository name in GitHub: https://github.com/openedx/xblock-drag-and-drop-v2/

#. **Python module name:** The python module name appears in the path of
   XBlock translations in the `openedx-translations repo`_. For example,
   the Drag and Drop XBlock will have ``drag_and_drop_v2`` python module name
   in the translations directory structure::

     translations/xblock-drag-and-drop-v2/drag_and_drop_v2/conf/locale/...

The ``pull_xblock_translations`` command will use the Python module name
instead of the repository name to pull translations from the
`openedx-translations repo`_ via ``atlas``.

> This trade-off has the following pros and cons:

Reviewer: ```suggestion
Using the Python module name has the following pros and cons:
```

**Pros:**

- The python module name is available without needing to install the XBlock,
  or parse the ``setup.py`` file.
- It is available in Python runtime.
- It is available in the `openedx-translations repo`_
  file structure.
- It is unique in the virtual environment which prevents
  collisions.
- The python module name of XBlocks doesn't change often if at all.

**Cons:**

- The python module name can be confused as the XBlock tag, which can
  be different in some XBlocks.
- The unique and stable identifier of XBlocks is the tag, not the
  python module name. Therefore, this decision will implicitly make
  the python module name another unique identifier for XBlocks.

The trade-offs are acceptable and this decision is reversible in case

> the ``xblock.tag`` needed to be used. However, this will require parsing

Reviewer: ```suggestion
the ``xblock.tag`` needs to be used. However, this will require parsing
```
